### PR TITLE
Unit tests for post-quantum groups.

### PR DESCRIPTION
**Pull request by anhu**
Also, fixes for the things they caught such as:

- ssl->arrays->preMasterSecret is pre-allocated so copy into it instead of
  moving ownership of buffer.
- server does not need to save the public key.
- in TLSX_KeyShare_Parse() don't call TLSX_KeyShare_Use() because its done in
  TLSX_PopulateExtensions().
- in TLSX_KeyShare_Use(), the server generates the ciphertext while the client
  generates the public key.
- in TLSX_PopulateExtensions(), prevent client from calling TLSX_KeyShare_Use()
  because its already been done.
- Support longer curve/group names.

### CODE REVIEW

File: src/tls.c [hunk 1]
```diff
@@ -7306,12 +7306,16 @@ static int TLSX_KeyShare_ProcessOqs(WOLFSSL* ssl, KeyShareEntry* keyShareEntry)
...
         /* I am the server, the shared secret has already been generated and
-         * is in keyShareEntry->ke; transfer ownership of the buffer. */
-        if (ssl->arrays->preMasterSecret != NULL)
-            XFREE(ssl->arrays->preMasterSecret, ssl->heap,
-                  DYNAMIC_TYPE_SECRET);
-        ssl->arrays->preMasterSecret = keyShareEntry->ke;
+         * is in keyShareEntry->ke; copy it to the pre-master secret
+         * pre-allocated buffer. */
+        if (keyShareEntry->keLen > ENCRYPT_LEN) {
+            WOLFSSL_MSG("shared secret is too long.\n");
```
Comment (dgarske) [inline src/tls.c:7312]: Old code free'd `XFREE(ssl->arrays->preMasterSecret, ssl->heap, DYNAMIC_TYPE_SECRET);`.  Also the WOLFSSL_MSG already adds `\n`.

File: src/tls.c [hunk 2]
```diff
@@ -7493,16 +7503,29 @@ static int TLSX_KeyShareEntry_Parse(WOLFSSL* ssl, const byte* input,
...
-    if (ke == NULL)
-        return MEMORY_E;
-    XMEMCPY(ke, &input[offset], keLen);
+#ifdef HAVE_LIBOQS
+    if (group >= WOLFSSL_OQS_MIN &&
+        group <= WOLFSSL_OQS_MAX &&
+        ssl->options.side == WOLFSSL_SERVER_END) {
+        /* For KEMs, the public key is not stored. Casting away const because
+         * we know for KEMs, it will be read-only.*/
+        ke = (byte *) &input[offset];
```
Comment (dgarske) [inline src/tls.c:7512]: Can you remove the spaces o the cast `(byte*)&input[offset];`

**Review by dgarske — CHANGES_REQUESTED** [review]
Couple minor comments.

File: src/tls.c (hunk 1 above)
Comment (anhu) [reply src/tls.c:7312]: fixed here and below.

File: src/tls.c (hunk 2 above)
Comment (anhu) [reply src/tls.c:7512]: done

**Review by dgarske — APPROVED** [review]

**Pull request merged**